### PR TITLE
feat: delete device

### LINKED ISSUE #9: Use case for deleting a device
Devices are those who can trigger the use cases related with reminders, those could be an NFC tag, bluetooth device, etc, etc, etc.

This use case should:
1) Given the deviceId, search if the device exists (depends on #8 ), if not throw an exception (NoSuchDeviceException)
2) If the device exists, remove it from the repository.

### PR DESCRIPTION
**Pull request by oriolagobat**
Closes #9

### CODE REVIEW

File: src/main/kotlin/device/application/DeviceSearcher.kt [hunk 1]
```diff
@@ -12,6 +12,6 @@ class DeviceSearcher(
      * @see Device
      * @return null or a device instance if it exists
      */
-    fun get(identifier: String) = get(Identifier(identifier))
-    private fun get(identifier: Identifier) = repository.search(identifier)
+    fun search(identifier: String) = search(Identifier(identifier))
```
Comment (Pablito2020) [inline src/main/kotlin/device/application/DeviceSearcher.kt:15]: I won't close this pull request but, for the next ones, be aware that here you are doing a refactor (that isn't a must-to-have in this use case*) + adding a feature. This breaks the statements discussed on the project rules ( #43  ).

The next time, if you can, do two different pull requests (one for the refactor and another for the feature). 

* I didn't mean that this refactor was unnecessary, I mean that, in the context of adding the "remove device" feature wasn't necessary and could be implemented on another pull request.

**Review by FerranAD — APPROVED** [review]

File: src/main/kotlin/device/application/DeviceSearcher.kt (hunk 1 above)
Comment (oriolagobat) [reply src/main/kotlin/device/application/DeviceSearcher.kt:15]: You're right, I'll do it that way from now on.

File: src/test/kotlin/device/application/DeviceRemoverTest.kt [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        Mockito.verify(repository, Mockito.times(0)).remove(id)
+    }
+
+    @Test
+    fun `Existing device gets deleted from the repository`() {
+        val device = DeviceMother.getValidDevice()
+        val primitiveId = IdentifierMother.getPrimitiveFrom(device.id)
+        Mockito.`when`(repository.search(device.id)).thenReturn(device)
+
+        assertDoesNotThrow { useCase.remove(primitiveId) }
```
Comment (Pablito2020) [inline src/test/kotlin/device/application/DeviceRemoverTest.kt:42]: The assertion isn't needed, if it throws an exception the test will fail anyway.
```suggestion
        useCase.remove(primitiveId)
```

**Review by Pablito2020 — CHANGES_REQUESTED** [review]

**Review by Pablito2020 — APPROVED** [review]

**Comment by Pablito2020** [comment]
@FerranAD if you merge it, do squash

**Review by FerranAD — APPROVED** [review]

**Pull request merged**